### PR TITLE
Add social panel

### LINKED ISSUE #83: Add Social Buttons to our Front Page
A request from marketing. Add the social buttons somewhere in the front page, ideally in the side panel above currently playing show: 
image

To add to the sidebar area, checkout `Frontpage.jsx`, looking at the `FrontpageContent` component. Styling is in `frontpage.scss`. 

Here are the following social media link we would like to add: 
Facebook: https://www.facebook.com/uclaradio
Twitter: https://twitter.com/UCLAradio
Instagram: https://www.instagram.com/uclaradio/
Snapchat: https://www.snapchat.com/add/uclaradio

Please use [FontAwesome](http://fontawesome.io/) as that's what we are already using.

### PR DESCRIPTION
**Pull request by xualexander**
## Types of changes

- [ ] Bugfix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Refactor (change which changes the codebase without affecting its external behavior)
- [ ] Non-breaking change (fix or feature that would causes existing functionality to work as expected)
- [ ] Breaking change (fix or feature that would cause existing functionality to __not__ work as expected)

## Purpose
Adds a social media link panel to the home page.
Resolves #83 
## Approach
Wrote a React component that generates Font Awesome icons as links to various social media pages, and added it to the FrontpageContent component. 
Uses an array of objects that stores the base icon name and URL for each link, in an effort to be more maintainable.
## Screenshot(s) 
screen shot 2017-11-03 at 17 21 31

## Checklist

- [x] My branch follows the branch naming scheme of UCLA Radio, and can merge into `master` without error.
- [x] My code follows the code style of this project, and I have linted it to confirm this.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

## Platform

__Operating System:__ macOS Sierra
__Node Version:__ 6.11.5

## Further Comments
Tested on my dev environment!

### CODE REVIEW

File: client/react/frontpage/components/SocialPanel.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    link: 'https://www.snapchat.com/add/uclaradio',
+  },
+];
+
+const SocialPanel = React.createClass({
+  convertToFaName(img, size) {
+    return 'fa fa-' + img + ' fa-' + size;
+  },
+  render() {
+    const self = this;
```
Comment (nathunsmitty) [inline client/react/frontpage/components/SocialPanel.js]: Why did you declare `self` here vs just using `this.convertToFaName` in line 38?

File: client/react/frontpage/components/SocialPanel.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return (
+      <div>
+        <p className="infoHeader">Social Media</p>
```
Comment (nathunsmitty) [inline client/react/frontpage/components/SocialPanel.js]: Hmm I'm not a big fan of the header saying "Social Media"—I feel like it's a little redundant. However, not having the banner also looks weird...

Lemme dig up some old mockups and see if I can suggest a better designed solution.

File: client/react/frontpage/components/SocialPanel.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        <div className="link-container">
+          {socialMediaLinks.map(function(obj) {
+            // change size here
+            let faName = self.convertToFaName(obj.img, '2x');
+            return (
+              <a href={obj.link} className="social-link">
```
Comment (nathunsmitty) [inline client/react/frontpage/components/SocialPanel.js]: You should check out the code in [ShowPage.js](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/components/ShowPage.js), [ShowPage.scss](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/components/ShowPage.scss), and [_common.scss](https://github.com/uclaradio/uclaradio/blob/master/client/react/frontpage/_common.scss). I implemented a similar feature for shows, and I think you could steal some styles from that.

Also, I really like how you implemented this with `.map`, so if you want to refactor ShowPage.js, that'd be 💯

File: client/react/frontpage/components/SocialPanel.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    img: 'instagram',
+    link: 'https://www.instagram.com/uclaradio/',
+  },
+  {
+    img: 'snapchat-ghost',
+    link: 'https://www.snapchat.com/add/uclaradio',
+  },
+];
+
+const SocialPanel = React.createClass({
```
Comment (binerys) [inline client/react/frontpage/components/SocialPanel.js]: Let's convert this component into a [stateless functional component](https://medium.com/@joshblack/stateless-components-in-react-0-14-f9798f8b992d). I linked to a medium post that describes what it is and an example of how to do it. Essentially, our component is not holding any state and simply needs to return markup. 

If you end up following the guide I linked, don't worry about the part where he mentions `propTypes`. We don't have that yet. Let me know if you have any questions (especially with all the ES6 arrow function stuff).

File: client/react/frontpage/components/SocialPanel.js (hunk 3 above)
Comment (binerys) [reply client/react/frontpage/components/SocialPanel.js]: +1 to your use of map to produce this social media panel. I'd love to see your Social Media panel utilized in our ShowPage component.

File: client/react/frontpage/components/SocialPanel.js (hunk 1 above)
Comment (xualexander) [reply client/react/frontpage/components/SocialPanel.js]: Oops, was just confused about binding in React. Fixed.

**Comment by xualexander** [comment]
I refactored SocialPanel and updated Frontpage.js to show how we could reuse the component for ShowPage and ShowBlurb. However, the social media links for each show are currently stored in an object that requires checking whether the social media exists whenever social media icons are rendered.

If we change the social media object(s) for each show to be an array of objects (as shown in Frontpage.js), it will be more maintainable overall (if social media sites/icons change) and allow for cleaner reuse of SocialPanel. However, this would involve changing the Panel API. I was curious as to what you guys think, and how we should go about refactoring the API if this change is worthwhile.

File: client/react/frontpage/components/SocialPanel.js (hunk 2 above)
Comment (xualexander) [reply client/react/frontpage/components/SocialPanel.js]: Removed the header for now

File: client/react/frontpage/components/SocialPanel.js (hunk 4 above)
Comment (xualexander) [reply client/react/frontpage/components/SocialPanel.js]: Refactored!

File: client/react/frontpage/components/SocialPanel.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import './SocialPanel.scss';
+
+const SocialPanel = ({ links, size }) => (
+  <div className="link-container">
+    {links.map(function(obj) {
+      let faName = 'fa fa-' + obj.img + ' fa-' + size;
```
Comment (nathunsmitty) [inline client/react/frontpage/components/SocialPanel.js]: Prefer [string template literals](https://github.com/airbnb/javascript#es6-template-literals) over concatenation.

File: client/react/frontpage/components/SocialPanel.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import './SocialPanel.scss';
+
+const SocialPanel = ({ links, size }) => (
+  <div className="link-container">
+    {links.map(function(obj) {
```
Comment (nathunsmitty) [inline client/react/frontpage/components/SocialPanel.js]: I'd use an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) here.

**Comment by haejinjo** [comment]
I noticed there's no interaction when you hover over the buttons. It would be cool if they could change color or highlight or something for the *~*UX*~**~~

**Comment by binerys** [comment]
moving forward with merging this, great job! 👍

**Pull request merged**